src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs

```diff
@@ -836,8 +842,6 @@ internal readonly ImmutableArray<SyntaxNodeAnalyzerAction<TLanguageKindEnum>> Ge
                     builder.Add(syntaxNodeAction);
                 }
             }
-
-            return builder.ToImmutableAndFree();
```

Comment:
copies all over the place. we realy want to avoid that while doing our processing.